src/editor/widgets/comment/CommentWidget.jsx

```diff
@@ -89,36 +89,56 @@ const CommentWidget = props => {
 
   return (
     <>
-      { comments.map((body, idx) => 
-        <Comment 
-          key={idx} 
+      { comments.map((body, idx) =>
```

Comment:
I think it's not necessary to repeat the whole markup block. Instead, the if condition could be adapted accordingly. I think something like this should work (I'm splitting into two conditions for readability):

```js
// Pre-condition: will be true if the annotation exists, and Annotorious is not in read-only mode
const isReadable = (!props.readOnly && props.annotation);

// Extra condtion to: reply field exists if there is no comment yet, or disableReply is false.  
const hasReply = comments.length === 0 || !props.disableReply;
```

And then you can render the block if `{isReadble && hasReply &&  ...`. At least if I'm not making a logic mistake here.

Reply:
Thanks for the catch - that makes total sense and is changed